notebooks/mock-notebooks/login/User Login Flow.ipynb

```diff
@@ -0,0 +1,403 @@
+{
```

Comment:
<p>Just going to give some ideas here for potential scenarios that could be addressed:</p><ol><li>Error handling for when a domain is offline</li><li>Error handling for when a network is offline</li></ol><p> </p><p><strong>Ideas &amp; Questions:</strong></p><ul><li>Do we have a method that will return a list of available actions to run if so we can mention that method in the "Successfully Logged In" message. So a "Connecting to https://ca.openmined.com  Logging into Canada Domain...  Logged in 🥳!!  Logged in as sheldon@caltech.edu Please use the command: <code>command.name</code> at anytime to see a list of commands available" or something to that effect. Thoughts? </li><li>General question is it possible for the same user account to be logged into under 2 different IP's simultaneously?  </li><li>How does a user log out of a domain? Or is that not necessary in this case?</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OpenMined/PySyft/pull/6333/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Reply:
<p>Yes, these are some great edge cases, thanks Kyoko for pointing them out. I will add them.</p><p> </p><p>Ideas &amp; Questions:</p><ul><li>Yes we can definitely have something like <code>command.help</code> to list the available commands. Although, python provides a <code>help</code> method that can be used to accomplish a similar thing. So, <code>help(command)</code> list all the methods that can access through the command along with their usage. Possibly we can build something like <code>command.help</code> that executes 	 <code>help(command)</code> functionality . But yes, I will add this as part of the message.</li><li>By different IP's do you mean that can a user to login into two domains at the same time ?? If that the case then yes. A user can login into any number of domains at the same time.</li><li>Log out functionality in notebooks is not required, cause a user is logged in as long as the notebook is alive. When a notebook dies, a user can not access their logged  session. They  need to start a new one. Although, a similar think called auto session expiry can be something we can think of in future if there is a use case where we need to logout a user even from notebooks.</li></ul>

Reply:
<p>Thanks Shubham this helps clarify.^_^</p><ul><li>If there is already a Python function then no need to create a command ^_^</li><li>By different IP's I'm trying to see if there are edge cases or security vulnerabilities so by different IP's I mean a user on one device tries to login to say Sheldon's account on Canada Domain and a user on a different device also tries to login to Sheldon's account on Canada's domain. Is that currently possible to do?</li><li>As far as logging out thanks for clarifying wasn't sure of the natural behavior there ^_^</li></ul>

Reply:
@Kiaka007 Yes, a user's account can be accessed from two or more different devices/IPs simultaneously. Currently, there is no restriction.

Reply:
<p>Thanks @shubham there's nothing wrong with that by itself I just wanted to know so I can further think of edge cases ^_^ If you think of any just let me know ^_^ </p>